### PR TITLE
[GAL-402] Spacer migration

### PR DESCRIPTION
**Pull request by jakzaizzat**
**Changes**
- Migrated all the old `Spacer` component with the new `Stack` - https://github.com/gallery-so/gallery/pull/999 
- Remove `DepreciateSpacer` component
- Added `inline-flex`  & `flex-wrap` support to `Stack` component

### CODE REVIEW

File: src/components/core/Spacer/Stack.tsx [hunk 1]
```diff
@@ -21,8 +21,9 @@ export const HStack = styled.div<{
   justify?: CSSProperties['justifyContent'];
   grow?: boolean;
   shrink?: boolean;
+  inline?: boolean;
 }>`
-  display: flex;
+  display: ${({ inline }) => (inline ? 'inline-flex' : 'flex')};
```
Comment (jakzaizzat) [inline src/components/core/Spacer/Stack.tsx:26]: Added support for `inline-flex`

File: src/scenes/CommunityPage/CommunityPageView.tsx [hunk 2]
```diff
@@ -62,32 +62,52 @@ export default function CommunityPageView({ communityRef }: Props) {
...
-      </StyledListWrapper>
-      <DeprecatedSpacer height={64} />
+          </StyledHeader>
+          <VStack gap={isMobile ? 32 : 64}>
+            <VStack gap={isMobile ? 32 : 80}>
+              <Spacer />
+              <MemberListFilter />
+            </VStack>
+            <Spacer />
+          </VStack>
```
Comment (tbezman) [inline src/scenes/CommunityPage/CommunityPageView.tsx:99]: pretty sure we don't need two vstacks here. I think this is because of the Spacer misuse.

**Review by tbezman — COMMENTED** [review]
@jakzaizzat you are my hero. thank you so much for doing this!!!!

I think the only thing left we need to crush is the removal of most, if not all of the `Spacer` uses.

If you need spacing in the beginning of an element, then that element should have padding to push its children inwards. 

I know this is really tedious work that requires insane attention to detail and I really appreciate it.

**Comment by tbezman** [comment]
@jakzaizzat I also pushed up a few commits of some bugs  / changes I thought made sense

File: src/components/ManageWallets/ManageWallets.tsx [hunk 3]
```diff
@@ -80,36 +80,36 @@ function ManageWallets({ newAddress, queryRef }: Props) {
...
-          chain={wallet.chainAddress.chain}
-          setErrorMessage={setErrorMessage}
-          userSigninAddress={userSigninAddress}
-          setRemovedAddress={setRemovedAddress}
-          isOnlyWalletConnected={wallets.length === 1}
-        />
-      ))}
-      <DeprecatedSpacer height={16} />
+    <VStack gap={16}>
+      <Spacer />
```
[the remark this answers is not in the document]
Comment (jakzaizzat) [reply src/components/ManageWallets/ManageWallets.tsx]: Thanks @tbezman for the insight! Seems like I overly used the `<Spacer/>` component

File: src/components/core/Spacer/Stack.tsx [hunk 4]
```diff
@@ -21,15 +21,19 @@ export const HStack = styled.div<{
...
   gap: 0 ${({ gap }) => gap ?? 0}px;
 
   align-items: ${({ align }) => align ?? 'unset'};
   justify-content: ${({ justify }) => justify ?? 'unset'};
 
   flex-grow: ${({ grow }) => (grow ? '1' : 'unset')};
 
+  flex-wrap: ${({ wrap }) => wrap ?? 'unset'};
```
Comment (jakzaizzat) [inline src/components/core/Spacer/Stack.tsx:35]: Added `flex-wrap` support too

File: src/scenes/CommunityPage/CommunityPageView.tsx (hunk 2 above)
Comment (jakzaizzat) [reply src/scenes/CommunityPage/CommunityPageView.tsx]: Updated this block!

**Comment by jakzaizzat** [comment]
@tbezman Removed the unnecessary Spacer component! It was definitely misused of the component.

**Review by tbezman — APPROVED** [review]
LETS GOOOOOOOOOOOOOOOOOO

**Pull request merged**